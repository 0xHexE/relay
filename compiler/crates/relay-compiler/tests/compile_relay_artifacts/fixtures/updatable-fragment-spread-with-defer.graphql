query updatableFragmentSpreadWithDeferQuery {
  me {
    ...updatableFragmentSpreadWithDefer_user @defer
  }
}

fragment updatableFragmentSpreadWithDefer_user on User @updatable {
  firstName
}
