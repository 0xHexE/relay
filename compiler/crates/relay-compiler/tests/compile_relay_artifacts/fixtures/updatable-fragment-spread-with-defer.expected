==================================== INPUT ====================================
query updatableFragmentSpreadWithDeferQuery {
  me {
    ...updatableFragmentSpreadWithDefer_user @defer
  }
}

fragment updatableFragmentSpreadWithDefer_user on User @updatable {
  firstName
}
==================================== OUTPUT ===================================
{
  "fragment": {
    "argumentDefinitions": [],
    "kind": "Fragment",
    "metadata": null,
    "name": "updatableFragmentSpreadWithDeferQuery",
    "selections": [
      {
        "alias": null,
        "args": null,
        "concreteType": "User",
        "kind": "LinkedField",
        "name": "me",
        "plural": false,
        "selections": [
          {
            "kind": "Defer",
            "selections": [
              {
                "args": null,
                "kind": "FragmentSpread",
                "name": "updatableFragmentSpreadWithDefer_user"
              }
            ]
          }
        ],
        "storageKey": null
      }
    ],
    "type": "Query",
    "abstractKey": null
  },
  "kind": "Request",
  "operation": {
    "argumentDefinitions": [],
    "kind": "Operation",
    "name": "updatableFragmentSpreadWithDeferQuery",
    "selections": [
      {
        "alias": null,
        "args": null,
        "concreteType": "User",
        "kind": "LinkedField",
        "name": "me",
        "plural": false,
        "selections": [
          {
            "if": null,
            "kind": "Defer",
            "label": "updatableFragmentSpreadWithDeferQuery$defer$updatableFragmentSpreadWithDefer_user",
            "selections": [
              {
                "alias": null,
                "args": null,
                "kind": "ScalarField",
                "name": "firstName",
                "storageKey": null
              }
            ]
          },
          {
            "alias": null,
            "args": null,
            "kind": "ScalarField",
            "name": "id",
            "storageKey": null
          }
        ],
        "storageKey": null
      }
    ]
  },
  "params": {
    "id": null,
    "metadata": {},
    "name": "updatableFragmentSpreadWithDeferQuery",
    "operationKind": "query",
    "text": null
  }
}

QUERY:

query updatableFragmentSpreadWithDeferQuery {
  me {
    ...updatableFragmentSpreadWithDefer_user @defer(label: "updatableFragmentSpreadWithDeferQuery$defer$updatableFragmentSpreadWithDefer_user")
    id
  }
}

fragment updatableFragmentSpreadWithDefer_user on User {
  firstName
}


{
  "argumentDefinitions": [],
  "kind": "Fragment",
  "metadata": null,
  "name": "updatableFragmentSpreadWithDefer_user",
  "selections": [
    {
      "alias": null,
      "args": null,
      "kind": "ScalarField",
      "name": "firstName",
      "storageKey": null
    }
  ],
  "type": "User",
  "abstractKey": null
}
